pkg/commands/exec.go

```diff
@@ -0,0 +1,181 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				ioStreams.Error("Please specify an application name.")
+				return nil
+			}
+			argsLenAtDash := cmd.ArgsLenAtDash()
+			if argsLenAtDash != 1 {
+				ioStreams.Error("Please specify at least one commnad for the container.")
+				return nil
+			}
+
+			if err := o.Init(context.Background(), cmd, args); err != nil {
+				return err
+			}
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+			return nil
+		},
+		Annotations: map[string]string{
+			types.TagCommandType: types.TypeApp,
+		},
+	}
+	cmd.Flags().BoolVarP(&o.Stdin, "stdin", "i", defaultStdin, "Pass stdin to the container")
+	cmd.Flags().BoolVarP(&o.TTY, "tty", "t", defaultTTY, "Stdin is a TTY")
+	cmd.Flags().Duration(podRunningTimeoutFlag, defaultPodExecTimeout,
+		"The length of time (like 5s, 2m, or 3h, higher than zero) to wait until at least one pod is running",
+	)
+	return cmd
+}
+
+func (o *VelaExecOptions) Init(ctx context.Context, c *cobra.Command, argsIn []string) error {
+	o.Context = ctx
+	o.Cmd = c
+	o.Args = argsIn
+
+	env, err := GetEnv(o.Cmd)
+	if err != nil {
+		return err
+	}
+	app, err := application.Load(env.Name, o.Args[0])
+	if err != nil {
+		return err
+	}
+	o.Env = env
+	o.App = app
+
+	//TODO(roywang) currently it always get kubeconfig in default way
+	// but it should re-use kubeconfig from env
+	cf := genericclioptions.NewConfigFlags(true)
+	cf.Namespace = &o.Env.Namespace
+	o.f = k8scmdutil.NewFactory(k8scmdutil.NewMatchVersionFlags(cf))
+	return nil
+}
+
+func (o *VelaExecOptions) Complete() error {
+	compName, err := o.askComponent()
+	if err != nil {
+		return err
+	}
+	podName, err := o.getPodName(compName)
+	if err != nil {
+		return err
+	}
+	o.kcExecOptions.StreamOptions.Stdin = o.Stdin
+	o.kcExecOptions.StreamOptions.TTY = o.TTY
+
+	args := make([]string, len(o.Args))
+	copy(args, o.Args)
+	// args for kcExecOptions MUST be in such formart:
+	// [podName, COMMAND...]
+	args[0] = podName
+	return o.kcExecOptions.Complete(o.f, o.Cmd, args, 1)
+}
+
+func (o *VelaExecOptions) askComponent() (string, error) {
+	comps := o.App.GetComponents()
+	if len(comps) == 1 {
+		return comps[0], nil
+	}
+	prompt := &survey.Select{
+		Message: "You have multiple Components in your app. Please choose one component for logs: ",
+		Options: comps,
+	}
+	var compName string
+	err := survey.AskOne(prompt, &compName)
+	if err != nil {
+		return "", fmt.Errorf("choosing component name err %v", err)
+	}
+	return compName, nil
+}
+
+func (o *VelaExecOptions) getPodName(compName string) (string, error) {
+	clientSet, err := o.f.KubernetesClientSet()
+	if err != nil {
+		return "", err
+	}
+	podList, err := clientSet.CoreV1().Pods(o.Env.Namespace).List(o.Context, v1.ListOptions{
+		LabelSelector: labels.Set(map[string]string{
+			oam.LabelAppName:      o.App.Name,
+			oam.LabelAppComponent: compName,
```

Comment:
I'm not sure if the pod will have these labels?

Reply:
It depends on how Workload controller works, e.g., PodSpecWorkload will always pass these labels to PodTemplate 
But we cannot make sure other Workloads work in the same way.
Let me mark TODO here and enhance it in a rigorous manner later.